src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -1716,5 +1684,85 @@ private static bool ConvertExceptionMessageFromStringToMessageObject(JObject exc
 
             return modifiedNotification;
         }
+
+        private delegate void ActionOnJObject(JObject jObject);
+
+        private static void ApplyActionOnNodeIfPathExists(string[] possibleNodePaths, JObject rootNode, ActionOnJObject action)
+        {
+            var nodesToUpdate = FindAllNodes(possibleNodePaths, rootNode);            
+
+            foreach (JObject node in nodesToUpdate)
+            {
+                action(node);
```

Comment:
>action(node); [](start = 16, length = 13)

I'd suggest applying the action on each possible path. This approach maximizes memory consumption.